src/pages/scripting/how-it-works/index.md

```diff
@@ -0,0 +1,100 @@
+    reject("Message");
+});
+``` 
+
+## User Interface
+Scripts can only show a dialog UI. Any UI created by a script is modal in nature and must use **global await** or it will be destroyed when the script is done running. Photoshop automatically shows a **progress bar** if the script takes more than 2-3 seconds to finish.
+
+
+### Global Await
+`await` expressions cause `async` function execution to pause until a Promise is either fulfilled or rejected, and to resume execution
+of the `async` function after fulfillment. When resumed, the value of the `await` expression is that of the fulfilled Promise.
+
+Global await means awaiting for a global scope async function to finish. 
+
+```js
+function anyAsyncFunction() {
+    return new Promise((resolve, reject) => {
+        const val = Math.random();
+        if (val > 0.5) {
+            console.log("Resolve promise");
+            resolve(val);
+        } else {
+            console.error("Rejecting promise");
+            reject(val);
+        }
+    })
+}
+
+// Global await
+try {
+    const val = await anyAsyncFunction();
+    console.log("Value from function", val);
+} catch {
+    console.log("Rejected promise!;")
+}
+console.log("Script completed");
+```
+
+## Permitted UXP Modules
+With plugin development, developers define which UXP modules they want to access in the `manifest.json` file. Script permissions are managed by Photoshop internally and no manifest configuration is required. 
+
+**At this time, not all UXP modules are accessible by scripts but we plan on enabling more modules in future UXP versions.**  All modules that are currently supported are enabled by default:
+
+
+| UXP Module | Supported | Current Access Level | Sample
+| --- | --- | --- | --- | 
+| Fonts | Yes | Installed fonts can be read | [View sample.](../samples/index.md#access-installed-fonts) |
+| Clipboard | Yes | Read/write access | [View sample.](../samples/index.md#readwrite-to-clipboard)| 
+| LocalFileSystem | Yes | File pickers can be used to open/save files. You can also call the ```getTemporaryFolder API``` to access a temporary data folder. | [View sample.](../samples/index.md#access-the-local-filesystem) |
+| Network | No |  |
+| LaunchProcess | No |  |
+| WebView | No |  |
+| IPC  | No |  |
+
+If you need to use the UXP modules that are not yet enabled for scripts, you should opt to create a UXP plugin instead. 
+
+## FAQ
+* What are the required minimum versions for UDT (for debugging) and Photoshop?
+  * UXP Scripting is available in Photoshop v23.5 and UDT v1.6.
+* Can you invoke a script from within another script? 
+  * No, inter-script communication is not possible. 
+* Can you pass arguments to scripts? 
+  * For this first release, passing arguments is not supported but **will be possible in a future release.**
+* Can scripts be executed from plugins? 
+  * No; however, any UXP script code should be able to run from within a UXP plugin.
```

Comment:
Caveat: point to the permissions differences.  
You can run it, but it could fail do to differences in permissions.